back/src/forms/validation-helpers.ts

```diff
@@ -0,0 +1,53 @@
+import { parse } from "date-fns";
+import { string } from "yup";
+
+const allowed_formats = [
+  "yyyy-MM-dd",
+  "yyyy-MM-dd'T'HH:mm:ss",
+  "yyyy-MM-dd'T'HH:mm:ssX",
+  "yyyy-MM-dd'T'HH:mm:ss.SSS",
+  "yyyy-MM-dd'T'HH:mm:ss.SSSX"
+];
+
+/**
+ * Check an incoming string is a date formatted according to allowed_formats
+ * "2020-11-23", "2020-11-23T13:34:55","2020-11-23T13:34:55Z", "2020-11-23T13:34:55.987", "2020-11-23T13:34:55.987Z"
+ */
+const isValidDatetime = str => {
+  if (!str) {
+    return true;
+  }
+  for (let fmt of allowed_formats) {
```

Comment:
j'ai tslint qui se plaint ici "Identifier 'fmt' is never reassigned; use 'const' instead of 'let'"